### PR TITLE
Reduce monkeypatching in pyflakes checker

### PR DESCRIPTION
**Pull request by tomasr8**
Chipping away at #183 

Got rid of the monkeypatched `LAMBDA`.

What's changed ->
- Default argument values cannot be forward references anymore:

```python
def bar(x: type = SomeClass) -> None: ...  # F821 undefined name 'SomeClass'
class SomeClass: ...
```
I think it makes sense to not allow this anyway - it's similar to what we did with classes previously.

### CODE REVIEW

**Review by srittau — APPROVED** [review]
I'm fine with this change, but I'll wait for a second opinion before merging.

**Review by AlexWaygood — APPROVED** [review]
LGTM, thank you!

**Comment by AlexWaygood** [comment]
I confirmed locally that there are no new hits on typeshed from this PR, even if F821 is reenabled in typeshed's `.flake8` file.

**Pull request merged**
